test failure: unit_test.FrontmatterTest testMethod=test_pretty_dumping

```
============================================================================ test session starts =============================================================================
platform linux -- Python 3.10.10, pytest-7.3.1, pluggy-1.0.0
rootdir: /tmp/makepkg/python-frontmatter/src/python-frontmatter-1.0.0
collected 59 items

tests/test_docs.py ...                                                                                                                                                 [  5%]
tests/test_files.py ......................                                                                                                                             [ 42%]
tests/unit_test.py .......F..............s....s....s.                                                                                                                  [100%]

================================================================================== FAILURES ==================================================================================
____________________________________________________________________ FrontmatterTest.test_pretty_dumping _____________________________________________________________________

self = <unit_test.FrontmatterTest testMethod=test_pretty_dumping>

    def test_pretty_dumping(self):
        "Use pyaml to dump nicer"
        # pyaml only runs on 2.7 and above
        if pyaml is not None:

            with codecs.open("tests/yaml/unpretty.md", "r", "utf-8") as f:
                data = f.read()

            post = frontmatter.load("tests/yaml/unpretty.md")
            yaml = pyaml.dump(post.metadata)

            # the unsafe dumper gives you nicer output, for times you want that
            dump = frontmatter.dumps(post, Dumper=pyaml.UnsafePrettyYAMLDumper)

>           self.assertEqual(dump, data)
E           AssertionError: '---\ndestination:\n  encoding:\n    xz:\[460 chars]der.' != "---\ndestination:\n  encoding:\n    xz:\[464 chars]der."
E             ---
E             destination:
E               encoding:
E                 xz:
E                   enabled: true
E                   min_size: 5120
E                   options:
E                   path_filter:
E               result:
E                 append_to_file:
E                 append_to_lafs_dir:
E                 print_to_stdout: true
E               url: http://localhost:3456/uri
E             filter:
E           -   - Длинный стринг на русском
E           +   - 'Длинный стринг на русском'
E           ?     +                         +
E           -   - Еще одна длинная строка
E           +   - 'Еще одна длинная строка'
E           ?     +                       +
E             ---
E
E             This is a test of both unicode and prettier dumping. The above metadata comes from the [pretty-yaml](https://github.com/mk-fg/pretty-yaml) readme file.
E
E             Metadata should be dumped in order.

tests/unit_test.py:130: AssertionError
========================================================================== short test summary info ===========================================================================
FAILED tests/unit_test.py::FrontmatterTest::test_pretty_dumping - AssertionError: '---\ndestination:\n  encoding:\n    xz:\[460 chars]der.' != "---\ndestination:\n  encoding:\n    xz:\[464 chars]der."
================================================================== 1 failed, 55 passed, 3 skipped in 0.18s ===================================================================
```

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python ${{ matrix.python-version }}

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -11,12 +11,12 @@ jobs:
       matrix:
         python-version: ["3.8", "3.9", "3.10", "3.11"]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         name: Configure pip caching
         with:
           path: ~/.cache/pip
@@ -32,12 +32,12 @@ jobs:
     runs-on: ubuntu-latest
     needs: [test]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
-          python-version: "3.8"
-      - uses: actions/cache@v2
+          python-version: "3.11"
+      - uses: actions/cache@v3
         name: Configure pip caching
         with:
           path: ~/.cache/pip

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -9,9 +9,9 @@ jobs:
       matrix:
         python-version: ["3.8", "3.9", "3.10", "3.11"]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
       - uses: actions/cache@v2

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -7,7 +7,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "3.11-dev"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python ${{ matrix.python-version }}

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -14,7 +14,7 @@ jobs:
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
-      - uses: actions/cache@v2
+      - uses: actions/cache@v3
         name: Configure pip caching
         with:
           path: ~/.cache/pip

--- a/setup.py
+++ b/setup.py
@@ -37,11 +37,10 @@ setup(
         "License :: OSI Approved :: MIT License",
         "Natural Language :: English",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.6",
-        "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
     ],
     test_suite="test",
 )

--- a/tests/test_files.py
+++ b/tests/test_files.py
@@ -2,7 +2,7 @@
 Test individual files with frontmatter against expected results.
 Files should be in a subdirectory under `tests`, usually sorted by format (yaml, toml, json).
 
-For a file called hello-world.markdown, there should be a corresponding file called hello-world.json
+For a file called hello-world.markdown, there should be a corresponding file called hello-world.result.json
 matching the expected output.
 """
 import os
@@ -22,7 +22,7 @@ def files():
 
 
 def get_result_filename(path):
-    root, ext = os.path.splitext(path)
+    root, _ = os.path.splitext(path)
     return f"{root}.result.json"
 
 

--- a/tests/unit_test.py
+++ b/tests/unit_test.py
@@ -115,9 +115,7 @@ class FrontmatterTest(unittest.TestCase):
 
     def test_pretty_dumping(self):
         "Use pyaml to dump nicer"
-        # pyaml only runs on 2.7 and above
         if pyaml is not None:
-
             with codecs.open("tests/yaml/unpretty.md", "r", "utf-8") as f:
                 data = f.read()
 
@@ -125,10 +123,10 @@ class FrontmatterTest(unittest.TestCase):
             yaml = pyaml.dump(post.metadata)
 
             # the unsafe dumper gives you nicer output, for times you want that
-            dump = frontmatter.dumps(post, Dumper=pyaml.UnsafePrettyYAMLDumper)
+            dump = frontmatter.dumps(post, Dumper=pyaml.PYAMLDumper)
 
-            self.assertEqual(dump, data)
             self.assertTrue(yaml in dump)
+            self.assertEqual(dump, data)
 
     def test_with_crlf_string(self):
         markdown_bytes = b'---\r\ntitle: "my title"\r\ncontent_type: "post"\r\npublished: no\r\n---\r\n\r\nwrite your content in markdown here'

--- a/tests/yaml/unpretty.md
+++ b/tests/yaml/unpretty.md
@@ -2,14 +2,14 @@
 destination:
   encoding:
     xz:
-      enabled: true
+      enabled: yes
       min_size: 5120
       options:
       path_filter:
   result:
     append_to_file:
     append_to_lafs_dir:
-    print_to_stdout: true
+    print_to_stdout: yes
   url: http://localhost:3456/uri
 filter:
   - Длинный стринг на русском

